### ISSUE TITLE
Incorrect lastversion for a specific repo

### ISSUE THREAD

**Issue by alexbelgium** [dvershinin/lastversion#55]
Hello,
Thanks very much for you app! I'm using the lastest version and have an issue.

For portainer/portainer, it retrieves a lastversion of 2.6.2. However, even though this version is out, it is not the "latest" release which is 2.6.1 as shown here : https://github.com/portainer/portainer/releases/latest

Is this normal behavior? Can I avoid it?

Thanks!

Edit : I call the app with 
```
lastversion "https://github.com/portainer/portainer"
```

**Comment by dvershinin** [comment]
It is normal behavior. This is what `lastversion` is for, really. In this particular case, the repo maintainer forgot/did not file a formal release for the latest tag via GitHub UI. `lastversion` correctly reports 2.6.2 as the latest release, which it really is.

**Comment by dvershinin** [comment]
P.S. I looked at the related issue. So you probably want something as "latest release having assets", or "latest formal release". I was going to add something like this to `lastversion`. Will keep the ticket open for that.

**Comment by alexbelgium** [comment]
Thanks very much. 

I've built an addon for [home assistant](https://www.home-assistant.io/) around your app, to automatically update addons based on latest github releases. I works perfectly for many addons. Thanks for having developed this and the very reactive support

**Comment by dvershinin** [comment]
@alexbelgium please try out lastversion v1.5.0.

You can run the following for formal release only:

```bash
lastversion https://github.com/portainer/portainer  --having-asset
```

Can be tested on a test repo:

```bash
lastversion https://github.com/lastversion-test-repos/portainer  --having-asset
```

My own application of this is slightly different than yours, as I'm checking [Telegram releases](https://github.com/telegramdesktop/tdesktop/releases) for specific asset "Linux 64 bit: Binary" (and they are nice to file releases with this same asset name) so I can now do:

```bash
lastversion telegramdesktop/tdesktop --having-asset "Linux 64 bit: Binary"
```

(sometimes they are late in uploading this asset, this is where this is useful, to not erroneously get the latest releases without this asset).

But in your case, there is still space for error, if portainer files a formal release but does not upload any asset to it. with --having-asset lastversion will still get that release as latest, because technically it is formal and has assets (source). But it might not have any actual assets you are interested in, and you can't use `--having-asset` with value, because their assets do not have fixed name.

**Comment by alexbelgium** [comment]
Thanks! I'll test it and report soon. Does it accepts * in the asset name? If yes then the portainer names are standard as only the number changes :) thanks again!

**Comment by dvershinin** [comment]
Pardon for confusing us both, disregard my prior message. 

Released another version 1.5.1 with consistent behavior for it. Documented [here](https://lastversion.getpagespeed.com/#use-case-releases-with-specific-assets).

`--having-asset` without value will act in the way you want. If there's an official/formal release with source only, it won't match.

So for your intention, simply have it like this:

```
lastversion "https://github.com/portainer/portainer" --having-asset
```

**Comment by alexbelgium** [comment]
This seems to work great! Thanks very much!
